worlds/sc2/rules.py

```diff
@@ -487,6 +487,63 @@ def all_in_requirement(self, state: CollectionState):
                 and state.has_any({item_names.HIVE_MIND_EMULATOR, item_names.PSI_DISRUPTER, item_names.MISSILE_TURRET}, self.player)
             )
 
+    def all_in_z_requirement(self, state: CollectionState):
+        """
+        All-in (Zerg)
+        :param state:
+        :return:
+        """
+        beats_kerrigan = (
+                state.has_any({item_names.ZERGLING, item_names.MUTALISK, item_names.INFESTED_MARINE}, self.player)
+                or state.has_all({item_names.SWARM_HOST, item_names.SWARM_HOST_RESOURCE_EFFICIENCY}, self.player)
+                or self.morph_brood_lord(state)
+                or self.advanced_tactics
+        )
+        if get_option_value(self.world, 'all_in_map') == AllInMap.option_ground:
+            # Ground
+            defense_rating = self.zerg_defense_rating(state, True, False)
+            if state.has_any({item_names.MUTALISK, item_names.INFESTED_BANSHEE, item_names.INFESTED_DUSK_WINGS}, self.player) or self.morph_brood_lord(state):
+                defense_rating += 2
+            return defense_rating >= 13 and beats_kerrigan
+        else:
+            # Air
+            defense_rating = self.zerg_defense_rating(state, True, True)
+            return (
+                    defense_rating >= 9 and beats_kerrigan
+                    and state.has_any({item_names.MUTALISK, item_names.CORRUPTOR}, self.player)
+                    and state.has(item_names.SPORE_CRAWLER, self.player)
+            )
+
+    def all_in_p_requirement(self, state: CollectionState):
+        """
+        All-in (Protoss)
+        :param state:
+        :return:
+        """
+        beats_kerrigan = (
+                state.has_any({item_names.CENTURION, item_names.SENTINEL, item_names.SKIRMISHER,
+                               item_names.CARRIER, item_names.TRIREME, item_names.VANGUARD}, self.player)
```

Comment:
Kerrigan will one hit KO Carriers, the real dealing with her is having Feedback

Reply:
I forgot about Feedback, that adds a bunch of units to the list:
- High Templar
- Signifier w/ War Council
- a hypothetical "Can Make Archon" rule (the above plus DT and Ascendant) + High Archon
- a hypothetical "Can make Dark Archon" rule (Dark Archons or Merge DA with DT) + Dark Archon Feedback

Re: Kerrigan's Implosion, its cooldown is difficulty-dependent, and its shortest cooldown is 6 seconds; on top of that, I suspect she'll try to Razor Swarm any interceptors she sees before going for the implosion. I'll do some playtesting and if it's ridiculous then I'll remove Carrier/Trireme from the list.